src/chainstate/coordinator/mod.rs

```diff
@@ -142,19 +142,35 @@ impl From<DBError> for Error {
 }
 
 pub trait RewardSetProvider {
-    fn get_reward_set(&self, chainstate: &StacksChainState,
-                      pox_anchor_hash: &BlockHeaderHash, pox_anchor_consensus: &ConsensusHash) -> Result<Vec<StacksAddress>, Error>;
+    fn get_reward_set(&self, chainstate: &mut StacksChainState,
+                      burnchain: &Burnchain, sortdb: &SortitionDB, block_id: &StacksBlockId) -> Result<Vec<StacksAddress>, Error>;
 }
 
-pub struct PlaceholderRewardSetProvider();
-impl RewardSetProvider for PlaceholderRewardSetProvider {
-    fn get_reward_set(&self, _chainstate: &StacksChainState,
-                      _pox_anchor_hash: &BlockHeaderHash, _pox_anchor_consensus: &ConsensusHash) -> Result<Vec<StacksAddress>, Error> {
-        Ok(vec![])
+pub struct OnChainRewardSetProvider {
+
+}
+
+impl RewardSetProvider for OnChainRewardSetProvider {
+    fn get_reward_set(&self, chainstate: &mut StacksChainState,
+                      burnchain: &Burnchain, sortdb: &SortitionDB, block_id: &StacksBlockId) -> Result<Vec<StacksAddress>, Error> {
+        let btc_addresses = chainstate.get_reward_addresses(burnchain, sortdb, block_id)?;
+        let mut stx_addresses = vec![];
+        for ((version, hash), _) in btc_addresses {
+            stx_addresses.push(StacksAddress::new(version as u8, hash));
```

Comment:
I don't think this is right.  This `version` field is going to be one of the values for `AddressHashMode`.  They will need to be converted into one of the `C32_ADDRESS_VERSION_*` values, via either `AddressHashMode::to_version_mainnet()` or `AddressHashMode::to_version_testnet()`.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1874/commits/d35ef465e9fa2ce327a181117f8ca7933b9df075

Tests are green, but the CI is now failing on compiling a build for windows-64.